### DISCUSSION TITLE
参考文献.bib如果不在tex目录下无法正确显示文献信息

### DISCUSSION THREAD

**Discussion by dreistein543** [nju-lug/NJUThesis#123] [Q&A] [answered]
电脑系统 macOS 12.3.1
tex版本 Mactex 2022
编辑器 texstudio 4.2.2
模板使用方式 tex 安装 CTAN 库中 njuthesis 包（相关所有包均安装且升级到最新版）

相关命令

```latex
bib = {
style = author-year,
resource = {/reference/我的文库.bib}, % 将bib文件放在tex目录下的 reference文件夹中
},
```

报错信息形如
```latex
Latexmk: ====List of undefined refs and citations: Citation 'saidu2019' on page 1 undefined on input line 9 Citation 'scherer2020' on page 1 undefined on input line 12 Citation 'abulof2017' on page 1 undefined on input line 14 Citation 'yildiz2016' on page 1 undefined on input line 14 Citation 'bitektine2015' on page 1 undefined on input line 14 Citation 'dingjianping2017' on page 1 undefined on input line 15
```

不知道是我没有正确指定bib文件位置（应该不可能，在文末指定`\bibliography{\reference\我的文库.bib}` texstudio可以正确读取bib文件内容）还是有什么其他因素导致的？

另外，如果 bib 文件不在 tex 目录下，似乎通过指定 `\bibliograhy{}` 的方式来 cite while you write 时（输入`\cite{...}` 或 `\textcite{...}` ），texstudio也无法正常预览待插入文献列表，不知道两者是否有关联，因为bib在tex目录下时，可以正常预览 。（P.S. 我知道指定 `\bibliography{}` 会无法正确编译，所以只是写作时使用，编译前会注释掉）

**Comment by dreistein543** [comment, marked as answer]
已经知道问题所在了，应该是指定 `resource = {reference/我的文库.bib}`，在 `reference` 之前不需要 `/`

**Reply by atxy-blip** [reply]
`\bibliograhy` 是 BibTeX 的命令，请不要使用它…

> 指定 `resource = {reference/我的文库.bib}`

其实也可以用 `resource = {./reference/我的文库.bib}`，单独一个斜杠是无效的。此外，请尽量避免中文用户名。

**Reply by dreistein543** [reply]
使用 `\bibliography` 只是为了 `\cite` 时能自动补全，编译时会注释掉，不过这种方式还是过于繁琐，不知到有没有简单的方式实现 `\cite` 自动补全又不需要 `\bibliography`

**Reply by dreistein543** [reply]
我已经找到了自动补全命令的方式，并且在 #126 中详细介绍了使用方法
